src/midgard/CMakeLists.txt

```diff
@@ -1,14 +1,10 @@
 file(GLOB headers ${VALHALLA_SOURCE_DIR}/valhalla/midgard/*.h)
 
 set(sources
-  linesegment2.cc
```

Comment:
could we keep this out of headers and make explicit template instantiations or does that have another negative side effect that im not thinking of? i cant imagine there is any  other template agrument beyond double that we would care about